docs/docs/background-jobs.md

```diff
@@ -0,0 +1,670 @@
+
+## Overview & Quick Start
+
+Before we get into anything with jobs specifically, we want to make sure `NODE_ENV` is set properly in your dev environment as this tells the job workers where to pull your jobs from: either `api/src/jobs` (when in development) or `api/dist/jobs` (when in production-like environments).
+
+In your `.env` or `.env.defaults` file, add an entry for `NODE_ENV`:
+
+```.env
+NODE_ENV=development
+```
+
+If `NODE_ENV` is _not_ set then the job runner will assume you're running a production-like environment and get jobs from `api/dist/jobs`. If you try to start a job worker and see this error:
+
+```
+JobsLibNotFoundError: api/src/lib/jobs.ts not found. Run `yarn rw setup jobs` to create this file and configure background jobs
+```
+
+Then `NODE_ENV` is probably not set!
+
+### Workflow
+
+There are three components to the background job system in Redwood:
+
+1. Scheduling
+2. Storage
+3. Execution
+
+**Scheduling** is the main interface to background jobs from within your application code. This is where you tell the system to run a job at some point in the future, whether that's "as soon as possible" or to delay for an amount of time first, or to run at a specific datetime in the future. Scheduling is handled by calling `performLater()` on an instance of your job.
+
+**Storage** is necessary so that your jobs are decoupled from your application. By default jobs are stored in your database. This allows you to scale everything independently: the api server (which is scheduling jobs), the database (which is storing the jobs ready to be run), and the job workers (which are executing the jobs).
+
+**Execution** is handled by a job worker, which takes a job from storage, executes it, and then does something with the result, whether it was a success or failure.
+
+### Installation
+
+To get started with jobs, run the setup command:
+
+```bash
+yarn rw setup jobs
+```
+
+This will add a new model to your Prisma schema, and create a configuration file at `api/src/lib/jobs.js` (or `.ts` for a Typescript project). You'll need to run migrations in order to actually create the model in your database:
+
+```bash
+yarn rw prisma migrate dev
+```
+
+Let's look at the config file. Comments have been removed for brevity:
+
+```js
+import { PrismaAdapter, RedwoodJob } from '@redwoodjs/jobs'
+
+import { db } from 'src/lib/db'
+import { logger } from 'src/lib/logger'
+
+export const adapter = new PrismaAdapter({ db, logger })
+export { logger }
+
+export const workerConfig: WorkerConfig = {
+  maxAttempts: 24,
+  maxRuntime: 14_400,
+  sleepDelay: 5,
+  deleteFailedJobs: false,
+}
+
+RedwoodJob.config({ adapter, logger })
+
+export const jobs = {}
+```
+
+We'll go into more detail on this file later (see [RedwoodJob (Global) Configuration](#redwoodjob-global-configuration)), but what's there now is fine to get started creating a job.
+
+### Creating a Job
```

Comment:
I'll leave reviewing the rest of the docs for the moment until you've had a chance to update them.